src/core/custom-elements/rs-changelog.js

```diff
@@ -0,0 +1,85 @@
+// @ts-check
+/**
+ * Fetches commits between two commitish (tag or commit) - `from` and `to=HEAD`
+ * and shows them as a list of "changes during" those commitish. If a filter
+ * function is provided by the user, it is used to filter the commits that are
+ * to be shown. Otherwise, all commits are shown.
+ *
+ * @typedef {{message: string, hash: string}} Commit
+ */
+import { github } from "../github.js";
+import { hyperHTML } from "../import-maps.js";
+import { showInlineError } from "../utils.js";
+
+export const name = "rs-changelog";
+
+export default class ChangelogElement extends HTMLElement {
+  constructor() {
+    super();
+    this.props = {
+      from: this.getAttribute("from"),
+      to: this.getAttribute("to") || "HEAD",
+      /** @type {(commit: Commit) => boolean} */
+      filter:
+        typeof window[this.getAttribute("filter")] === "function"
+          ? window[this.getAttribute("filter")]
+          : () => true,
+    };
+  }
+
+  connectedCallback() {
+    const { from, to, filter } = this.props;
+    hyperHTML.bind(this)`
+      <ul>
+      ${{
+        any: fetchCommits(from, to, filter)
```

Comment:
Nit: this is still a bit of an eyesore for me... could we move it to a function?

Reply:
actually, never mind... it uses a lot of `this`.